File: python/entity.py
from __future__ import annotations
from random import randrange 

class Entity:
    def __init__(self, name) -> None:
        self.name = name
        self.hp = 100

    def deal_damage_to(self, entity: Entity, damage: int):
        entity.hp -= damage
        print("{0} Damages => {1} with {2} dmg".format(self.name, entity.name, damage))
    
    def assign_attack_damage(self, max: int) -> int:
        return randrange(max)

File: lua/entity.lua
local Entity = {}

function Entity.new(name)
    local self = setmetatable({}, { __index = Entity })
    self.name = name
    self.hp = 100
    return self
end

function Entity:deal_damage_to(entity, damage)
    entity.hp = entity.hp - damage
    print(string.format("%s Damages => %s with %d dmg", self.name, entity.name, damage));
    print("===============================")
end

function Entity:assign_attack_damage(max)
    return math.random(max)
end

return Entity
